internal/controller/management_controller.go

```diff
@@ -144,7 +138,49 @@ func (r *ManagementReconciler) Update(ctx context.Context, management *hmc.Manag
 	return ctrl.Result{}, nil
 }
 
-func (r *ManagementReconciler) Delete(_ context.Context, _ *hmc.Management) (ctrl.Result, error) {
+func (r *ManagementReconciler) Delete(ctx context.Context, management *hmc.Management) (ctrl.Result, error) {
+	l := log.FromContext(ctx)
+	l.Info("Removing HelmReleases owned by HMC")
+	helmReleases := &v2.HelmReleaseList{}
+	listOpts := &client.ListOptions{
+		LabelSelector: labels.SelectorFromSet(map[string]string{hmc.HMCManagedLabelKey: "true"}),
+	}
+	if err := r.Client.List(ctx, helmReleases, listOpts); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	var errs error
+	for _, hr := range helmReleases.Items {
+		lc := l.WithValues("name", hr.Name, "namespace", hr.Namespace)
+		lc.Info("Processing HelmRelease removal")
+		if hr.ObjectMeta.DeletionTimestamp.IsZero() {
+			if hr.Name == management.Spec.Core.HMC.HelmReleaseName() {
+				lc.Info("Suspending HMC HelmRelease")
+				// This one has to be removed manually. HelmRelease is marked as suspended before deletion.
+				hr.Spec.Suspend = true
+				if err := r.Client.Update(ctx, &hr); err != nil {
```

Comment:
Maybe skip the update if `hr.Spec.Suspend` is already set?

Reply:
yep, makes sense